adam_preprocessing/singlefile_stroke_extraction.py

```diff
@@ -32,13 +32,26 @@ def main():
         default="dummy",
         help="String label to use."
     )
+    parser.add_argument(
+        "--merge-small-strokes",
+        action="store_true",
+        help="Merge small strokes before filtering."
+    )
+    parser.add_argument(
+        "--no-merge-small-strokes",
+        action="store_false",
+        dest="merge_small_strokes",
+        help="Don't merge small strokes before filtering."
+    )
```

Comment:
It's so that you can disable the behavior when using a bash script that passes `--merge-small-strokes`. This lets you pass `--no-merge-small-strokes` via the `"$@"` args. In this case I don't expect we'll need to, but I've run into this case of wanting to override the defaults before with scripts that pass lots of boolean defaults, specifically Hugging Face model training/inference scripts. With the Hugging Face argument parser you can pass an explicit boolean value `--merge-small-strokes True` or `--merge-small-strokes False` but argparse doesn't support that i don't think.

For example, say you are invoking stroke extraction from a bash script. You pass `--merge-small-strokes` as part of the bash script. You also pass any leftover bash args to the Python script. If you want to invoke the bash script but disable merging for whatever purpose, the `store_true` alone isn't enough -- you'd have to either edit it out (works fine for debugging, but not as much for other things) or have a `store_false` flag you can pass to disable the behavior.

I'm fine with deleting this, though, because it doesn't seem necessary in this script.

Reply:
Ah makes sense. I've just not encountered a need to do such a thing :) -- happy to leave this in. I almost just approved and asked the question at the same time.